avm/res/dashboard/grafana/main.bicep

```diff
@@ -0,0 +1,184 @@
+metadata name = 'Azure Managed Grafana'
+metadata description = 'This module deploys an Azure Managed Grafana(Grafana).'
+metadata owner = 'Azure/module-maintainers'
+
+// imports
+import * as grafanaTypes from 'types/grafana.type.bicep'
+
+@description('Required. Name of your Azure Managed Grafana.')
+@minLength(2)
+@maxLength(23)
+param name string
+
+@description('Specifies the resource tags for the Azure Monitor managed service for Prometheus resource.')
+param tags object
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Specifies the api key setting of the Azure Managed Grafana resource.')
+@allowed([
+  'Disabled'
+  'Enabled'
+])
+param apiKey string = 'Disabled'
+
+@allowed([
+  'Essential'
+  'Standard'
+])
+@description('Specifies the sku of the Azure Managed Grafana resource.')
+param skuName string = 'Standard'
+
+@allowed([
+  'TenantReuse'
+])
+param autoGeneratedDomainNameLabelScope string = 'TenantReuse'
+param deterministicOutboundIP string = 'Disabled'
+param enterpriseConfigurations grafanaTypes.enterpriseConfigurationsType = {}
+param grafanaIntegrations grafanaTypes.grafanaIntegrationsType = {}
+
+param grafanaPlugins object = {}
+
+param grafanaMajorVersion string = '10'
+
+@description('Optional. Specifies the the state for enable or disable traffic over the public interface for the the Azure Managed Grafana resource.')
+@allowed([
+  'Disabled'
+  'Enabled'
+])
+param publicNetworkAccess string = 'Disabled'
+
+@description('Optional. The zone redundancy setting of the Azure Managed Grafana resource.')
+@allowed([
+  'Disabled'
+  'Enabled'
+])
+param zoneRedundancy string = 'Disabled'
```

Comment:
Shouldn't this be Enabled by default to be WAF-aligned?

Reply:
Note: Not all regions support that. I When this is enabled, I'd recommend to set the location in both the `defaults` & `waf-aligned` test cases to e.g., 'NorthEurope' using an implementation like here:
https://github.com/Azure/bicep-registry-modules/blob/4a59f7c495df0e9f51b214d89f9f75e44f8ec0ff/avm/res/db-for-my-sql/flexible-server/tests/e2e/max/main.test.bicep#L27-L29